lib/pbench/agent/utils.py

```diff
@@ -390,3 +393,60 @@ def collect_local_info(pbench_bin):
         hostdata[arg] = cp.stdout.strip() if cp.stdout is not None else ""
 
     return (version, seqno, sha1, hostdata)
+
+
+if os.environ.get("_PBENCH_UNIT_TESTS"):
+
+    class LocalRemoteHost:
+        """Mock out behavior for legacy unit tests.
+        """
+
+        def __init__(self):
+            self._local_names = frozenset(
+                [
+                    "127.0.0.1",
+                    "localhost",
+                    "172.16.42.42",
+                    "testhost.example.com",
+                    "testhost",
+                ]
+            )
+
+        def is_remote(self, host_name):
+            return host_name not in self._local_names
+
+
+else:
+
+    class LocalRemoteHost:
+        """Dynamically build a list of all the local IP addresses so that a simple
+        simple query interface provides the answer to the question, "is the given
+        host name or IP address remote?"
+        """
+
+        def __init__(self):
+            """Build up a list of local IP addresses from all the interfaces.
+            """
+            local_ips_l = []
+            for adapter in ifaddr.get_adapters():
+                for ip in adapter.ips:
+                    local_ips_l.append(ip.ip[0] if isinstance(ip.ip, tuple) else ip.ip)
+            self.local_ips = frozenset(local_ips_l)
+
+        def is_remote(self, host_name):
+            """Returns True if the given host name (or IP address) is remote else
+            returns False.
+            """
+            ips_l = []
+            try:
+                ip_addrs = socket.getaddrinfo(host_name, None)
+                for addrinfo in ip_addrs:
+                    ips_l.append(addrinfo[4][0])
+            except socket.gaierror:
```

Comment:
A quick look at the docs suggests that, other than `EAI_AGAIN`, these errors are either catastrophic or in the "should never happen" category for our usage, so it doesn't seem worthwhile to maintain `ip_address` as a fall-back (especially since it will only work if the input was numeric).  More to the point, it strikes me that this `except` block is going to eat errors that we would otherwise want to know about.

So, if you really want to maintain the fallback, I suggest that we should be be using it only if the `gaierror` is reporting an `EAI_AGAIN`.

Reply:
OK, I thought about this but I don't remember whether I forgot or just decided to be more conservative about my changes.